### PR TITLE
Refactor jrt buffer operations

### PR DESCRIPTION
**Pull request by akosthekiss**
`jrt_read_from_buffer_by_offset` is not used anywhere in the code
while `jrt_write_to_buffer_by_offset` is only used by snapshot
saving functions. Thus, this patch removes the read variant
completely and moves the write variant as a static function to
jerry.c. This empties out jrt.c, thus deleting.

JerryScript-DCO-1.0-Signed-off-by: Akos Kiss akiss@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/jerry.c [hunk 1]
```diff
@@ -1748,6 +1748,34 @@ typedef struct
...
+ * Note:
+ *      Offset is in-out and is incremented if the write operation completes successfully.
+ *
+ * @return true, if write was successful, i.e. offset + data_size doesn't exceed buffer size,
+ *         false - otherwise.
+ */
+static inline bool __attr_always_inline___
+snapshot_write_to_buffer_by_offset (uint8_t *buffer_p, /**< buffer */
+                               size_t buffer_size, /**< size of buffer */
+                               size_t *in_out_buffer_offset_p,  /**< [in,out] offset to write to
```
Comment (zherczeg) [inline jerry-core/jerry.c]: Alignment issue.

**Comment by akosthekiss** [comment]
fixed the style issue

**Comment by LaszloLango** [comment]
LGTM

File: jerry-core/jerry.c [hunk 2]
```diff
@@ -1748,6 +1748,34 @@ typedef struct
 } snapshot_globals_t;
 
 /**
+ * Write data to a specified buffer.
```
Comment (zherczeg) [inline jerry-core/jerry.c]: I would use "Write data into the specified buffer" or "Write data into a buffer". I know this is copied, but it sounds better to me.

**Comment by akosthekiss** [comment]
changed the wording and updated

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**